### DISCUSSION TITLE
UI should imply that freemods are applied on a per-item basis in the playlist song selection screen

### DISCUSSION THREAD

**Discussion by venix12** [ppy/osu#17281] [Ideas]
Currently, it seems like freemods are rather applied on a playlist-basis as there's no indication they  aren't (and quite sure it used to be? this caused some trouble when creating spotlights lobbies), therefore if you set them after adding the items the freemods will not apply on playlists while it might seem like they should. (also it's possible to set different freemods for different maps which might not be the intention while not displayed)

The soulution I'd suggest would is adding freemods indicator on each playlist item on song selection screen, similar to how it's done on regular mods.

![image](https://user-images.githubusercontent.com/30481900/158578646-0aab20d7-2250-44d2-8c78-17f5d4a6e4b8.png)

**Comment by smoogipoo** [comment]
Design: https://www.figma.com/file/XJCRjfvTlVIwZBWoAz5kB2/Client%2FPlaylist-Multiplayer?node-id=0%3A1

@arflyte The multiplayer beatmap panels currently don't display the freemods (only required mods). In the design, they still don't show it but there's a "freemod" section below the playlist.
1. I'm not sure on the freemod list being separate from the beatmap panel. It would be nice to be visible in the panel imo.
2. How does the user know which mods are currently active for them? Which are a combination of all required mods and some subset of the freemods.

**Comment by peppy** [comment]
As an aside, this was also part of my feedback from the large multiplayer match last week:

![Discord PTB 2022-03-17 at 05 41 51](https://user-images.githubusercontent.com/191335/158744587-16a98d9c-4a6b-42f2-bae3-346462c4bc6d.png)
